src/test/java/WebServerTest.java

```diff
@@ -11,25 +10,33 @@
     private int port = 1234;
     private MockServerSocketManager mockServerSocketManager;
     private RequestHandler requestHandler = new RequestHandler();
-    private CommunicationChannel mockServerCommunicationChannel = new MockServerCommunicationChannel();
     private ResponseHandler responseHandler = new ResponseHandler();
     private ServerStatus mockWebServerStatus = new MockWebServerStatus();
     private WebServer webServer;
 
-    @Before
-    public void setUp() throws IOException {
+    public void setUp(String request) throws IOException {
+        CommunicationChannel mockServerCommunicationChannel = new MockServerCommunicationChannel(request);
         mockServerSocketManager = new MockServerSocketManager(mockServerCommunicationChannel);
         webServer = new WebServer(mockServerSocketManager, requestHandler, responseHandler, mockWebServerStatus);
         webServer.start(port);
     }
 
     @Test
-    public void startWebServerListenForConnections() {
+    public void startWebServerListenForConnections() throws IOException {
+        setUp("GET /simple_get HTTP/1.1");
```

Comment:
I like that setup that is not necessary for the tests are abstracted away into this `setUp` method. However, I found that I was having to go and look at the method to figure out what the test was doing.

I wonder if a name change and perhaps a return type might help make this clearer?